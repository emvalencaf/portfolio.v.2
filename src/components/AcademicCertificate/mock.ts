import { AcademicCertificateProps } from ".";

export default {
	courseTitle: "Curso de React.Js e Next.Js (Intermediário e Avançado)",
	institution: "Udemy, Instrutor: Luiz Otávio Miranda",
	courseWork: "93 horas",
	startIn: 2932983,
	endIn: 3932983,
	urlDownload: "/assets/files/UC-44225967-b7ec-4add-8983-426364ad7bb6.pdf"
} as AcademicCertificateProps;