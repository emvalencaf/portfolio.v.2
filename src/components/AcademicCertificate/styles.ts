import styled, { css, DefaultTheme } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';

type WrapperAcademicCertificateProps = {
	theme: DefaultTheme;
	showDetails: boolean;
}

const changeCursor = () => css`
	cursor: pointer;
`;

export const Wrapper = styled.div<WrapperAcademicCertificateProps>`
	${({ theme, showDetails }) => css`
	padding: 2rem;
	${!showDetails && changeCursor()}

	& ${HeadingContainer} {
		margin: 0px;
	}
	`}
`;

export const HeaderAcademicCertificate = styled.div`
	${({ theme }) => css`
		display: flex;
		align-items: center;
		justify-content: center;
		margin: ${theme.spacings.small} auto;
	`}
`;

export const BodyAcademicCertificate = styled.div`
	${({ theme }) => css`
		display: flex;
		flex-direction: column;
		align-items: center;
		justify-content: center;
		gap: ${theme.spacings.small};
		margin: ${theme.spacings.small} auto;
	`}
`;