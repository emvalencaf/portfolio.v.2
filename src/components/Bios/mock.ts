import { BiosProps } from ".";

export default {
	bios: `
<p>
	Lorem ipsum dolor sit amet consectetur adipisicing elit. Inventore rem minus dolore ducimus quos, sed dolores odit maiores quam tempora atque deserunt voluptas perferendis, aspernatur, sequi dignissimos reiciendis. In, esse!
	Lorem ipsum, dolor sit amet consectetur adipisicing elit. Non a deleniti quia. Magnam aspernatur cumque, nemo vero libero excepturi consequuntur corporis ullam nobis molestiae? Atque nam praesentium ullam quaerat quisquam?
</p>
<p>
	Lorem ipsum dolor sit amet consectetur adipisicing elit. Optio cumque recusandae libero sequi, tempora iste, illo, possimus unde cum numquam quae corrupti minima tenetur mollitia earum repellendus animi molestiae voluptates.
	Lorem ipsum dolor sit, amet consectetur adipisicing elit. Obcaecati dolores fugiat ipsam ut odit sed aliquid ducimus molestiae vero ipsa nam veniam dolor libero labore qui pariatur, alias assumenda itaque?
	Lorem ipsum dolor sit amet, consectetur adipisicing elit. Illo cum temporibus nesciunt sequi odit ipsam harum magnam dolore a molestiae. Totam debitis, rerum aliquid sit non ratione. Et, omnis iusto!
</p>
<p>
	Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nam dicta aut cum sequi voluptatibus. Vel pariatur veritatis quidem. Culpa vero deleniti minima earum quod, odit deserunt fugit reprehenderit architecto in!
</p>
	`,
	profilePhoto: {
		altText: "profile picture",
		srcImg: "assets/images/potrait-photo.jpg",
	},
} as BiosProps;
