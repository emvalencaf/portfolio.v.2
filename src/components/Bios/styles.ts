import styled, { css } from "styled-components";
export const Wrapper = styled.div`
	${({ theme }) => css`
		@media ${theme.media.lteMedium} {
			flex-direction: column;
		}

		display: flex;
		flex-direction: row;
		flex-wrap: wrap;
		justify-content: space-between;
		align-items: center;
		gap: ${theme.spacings.xhero};
		text-align: justify;
		& p {
			text-align: justify;
			padding: ${theme.spacings.small};
			max-width: 550px;
		}
	`}
`;

export const ContentContainer = styled.div`
	${() => css`
		max-height: 350px;
		overflow-y: auto;
	`}
`;
