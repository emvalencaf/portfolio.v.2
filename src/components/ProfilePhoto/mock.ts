import { ProfilePhotoProps } from ".";

export default {
	srcImg: "assets/images/profile.png",
	altText: "website owner",
} as ProfilePhotoProps;
