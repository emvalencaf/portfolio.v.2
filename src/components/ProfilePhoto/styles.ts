import styled, { css } from 'styled-components';

export const Picture = styled.picture`
	${({ theme }) => css`
		& img  {
			width: ${theme.frameSizes.smallMedium};
			height: ${theme.frameSizes.smallMedium};
			border-radius: 20px;
			box-shadow: 20px 14px 20px 4px;

		}
	`}
`;
