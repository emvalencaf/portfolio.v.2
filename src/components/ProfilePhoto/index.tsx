// styles
import * as Styled from './styles';

// types
export type ProfilePhotoProps = {
	srcImg: string;
	altText: string;
};

const ProfilePhoto = ({ srcImg, altText }: ProfilePhotoProps) => {
	return (
		<Styled.Wrapper>
			<img src={srcImg} alt={altText} />
		</Styled.Wrapper>
	);
};

export default ProfilePhoto;
