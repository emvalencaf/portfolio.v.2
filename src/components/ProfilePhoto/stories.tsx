import { Meta, Story } from "@storybook/react/types-6-0";
import ProfilePhoto, { ProfilePhotoProps } from ".";
import mock from "./mock";

export default {
	title: "ProfilePhoto",
	component: ProfilePhoto,
	args: mock,
	argTypes: {
		altText: { type: "string" },
		srcImg: { type: "string" },
	},
} as Meta<ProfilePhotoProps>;

export const Template: Story<ProfilePhotoProps> = (args) => {
	return (
		<div>
			<ProfilePhoto {...args} />
		</div>
	);
};
