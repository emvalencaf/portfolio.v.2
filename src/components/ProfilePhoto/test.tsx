import { renderTheme } from "../../styles/render-theme";

// react component to be test
import ProfilePhoto from ".";

// mock
import mock from "./mock";

describe("<ProfilePhoto />", () => {
	it("should match a snapshot", () => {
		const { container } = renderTheme(<ProfilePhoto {...mock} />);

		expect(container).toMatchSnapshot();
	});
});
