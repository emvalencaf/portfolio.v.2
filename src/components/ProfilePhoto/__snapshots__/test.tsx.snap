// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfilePhoto /> should match a snapshot 1`] = `
<div>
  <div
    class=""
  >
    <img
      alt="website owner"
      src="assets/images/profile.png"
    />
  </div>
</div>
`;
